### PR TITLE
build: Mangle internal export names

### PR DESCRIPTION
**Pull request by shuding**
In #1962 we added a shared internal module for all subpath imports to use. However the named exports can't be minified (by many bundlers) because they're across modules. Here we are using TypeScript enums to ensure that 1) these names can be minified, and 2) we can still import the correct utils with typings easily.

This decreases the bundle size by 800+ bytes (minified) for just `'swr'`, and will probably save more if other subpaths are used such as `'swr/infinite'`.

### CODE REVIEW

File: core/index.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
+// @ts-nocheck
```
Comment (huozhi) [inline core/index.ts:1]: why do we need this?

Comment (shuding) [reply core/index.ts:1]: Good catch – it was for testing, removed!

File: _internal/enums.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+export const enum InternalUtils {
```
Comment (huozhi) [inline _internal/enums.ts:1]: maybe explain why we need to have this as comment here, to help contributors understand better this is for size compression purpose

**Comment by shuding** [comment]
Closing for now as most of bundlers are already handling this today. For the other cases that modules can’t be bundled together (bundleless solutions, or ending up in different chunks somehow), we will wait and see if it has an impact or not.

**Pull request closed without merging**